textx/metamodel.py

```diff
@@ -254,6 +257,10 @@ def __init__(self, file_name=None, classes=None, builtins=None,
 
     def register_scope_providers(self, sp):
         self.scope_providers = sp
+        for k in self.scope_providers.keys():
+            if isinstance(self.scope_providers[k], string_types):
+                self.scope_providers[k] = create_rrel_scope_provider(
+                    self.scope_providers[k])
```

Comment:
This could be:
```python
        for k, v in self.scope_providers.items():
            if isinstance(v, string_types):
                self.scope_providers[k] = create_rrel_scope_provider(v)

```

Reply:
code changed